api/src/main/java/org/apache/iceberg/transforms/Truncate.java

```diff
@@ -284,8 +284,30 @@ public boolean satisfiesOrderOf(Transform<?, ?> other) {
 
       if (predicate.isUnaryPredicate()) {
         return Expressions.predicate(predicate.op(), name);
-      } else if (predicate.isLiteralPredicate()) {
-        return ProjectionUtil.truncateArray(name, predicate.asLiteralPredicate(), this);
+      } else if (predicate instanceof BoundLiteralPredicate) {
```

Comment:
Why change this to `instanceof` instead of using `isLiteralPredicate`?

Reply:
That's a good call. I'm not 100% sure why this got changed. If memory serves, I think I might have just used this snippet as it was recommended earlier in the review process (which I think might have had the `instanceof` check). So I chose to err on the side of caution and not deviate from whatever was provided.

Going forward, I will use my own judgement about whether there's a better way to write something and consider that what is suggested might potentially be a reference (particularly in this case where the existing code is already using `isLiteralPredicate` and the original recommendation has clear intent). I agree that keeping `isLiteralPredicate` function is for sure a better idea.

I will update this 👍